pages/add/add.jsx

```diff
@@ -1,12 +1,201 @@
+    // There will be a query argument "loggedin" presented in the URL
+    // which is something we don't want it to stay at all times.
+    this.clearLoggedinParam();
+  },
+  clearLoggedinParam() {
+    let query = this.props.router.location.query;
+
+    // we don't need the "loggedin" param sent back from Pulse API to be presented at all times
+    delete query.loggedin;
+
+    browserHistory.replace({
+      pathname: this.props.router.location.pathname,
+      query: query
+    });
+  },
+  logOutUser(event) {
+    event.preventDefault();
+    Service.logout()
+      .then(() => {
+        // we have successfully logged user out
+        this.updateLoggedInInfo(false);
+        // redirect user to /featured page
+        browserHistory.push({
+          pathname: `/featured`
+        });
+      })
+      .catch(reason => {
+        console.error(reason);
+        // TODO:FIXME: what do we do here?
+      });
+  },
+  handleCreatorClick(event) {
+    event.preventDefault();
+    this.setState({numCreatorFields: this.state.numCreatorFields+1});
+  },
   render() {
+    let creatorFields = [];
+
+    for (var i=0; i<this.state.numCreatorFields; i++) {
+      creatorFields.push( <CreatorInputField key={i} />);
+    }
+
+    let ContentForUnauthedUser = ( <HintMessage imgSrc={`/assets/svg/icon-bookmark-selected.svg`}
+                                               header={`Sign in failed`}
+                                               btn={{to: `/featured`, text: `Explore featured`}}>
+                                    <p>Only Mozilla staff can login now as we test this new platform. Check back soon!</p>
+                                  </HintMessage>);
+
+    let ContentForAuthedUser = (
+      <div>
+        <h1>Share with the Network</h1>
+        <p>Do you have something to share? If it might be useful to someone in our network, share it here! Pulse includes links to products and software tools, research reports and findings, think pieces, white papers, interviews, and curricula. If it might be useful, share it … at any stage or fidelity.</p>
+        <form action="" method="post">
+          <h2>Basic Info</h2>
+          <div className="posted-by">
+            <p>Posted by:&nbsp;<span className="user-full-name">{UserData.username.get()}</span></p>
+            <p className="log-out-prompt">Not you? <button className="btn btn-link" onClick={this.logOutUser}>Sign out</button>.</p>
+          </div>
+          <fieldset hidden>
+            <label className="form-control-label">
+              Posted by:&nbsp;
+              <input name="" type="text" className="form-control" value="" readOnly />
+            </label>
+            <p>Not you? <a href="">Sign out.</a></p>
+          </fieldset>
+          <label className="form-control-label">
+            Title of the project<em className="required">(required)</em>
+            <input name="title" type="text" placeholder="Title" className="form-control" />
```

Comment:
if this is required, then this should have `required={"required"}`, so that the form can't be submitted until it has a value.